service/s3/s3manager/download.go

```diff
@@ -110,9 +126,10 @@ func NewDownloader(c client.ConfigProvider, options ...func(*Downloader)) *Downl
 //     })
 func NewDownloaderWithClient(svc s3iface.S3API, options ...func(*Downloader)) *Downloader {
 	d := &Downloader{
-		S3:          svc,
-		PartSize:    DefaultDownloadPartSize,
-		Concurrency: DefaultDownloadConcurrency,
+		S3:             svc,
+		PartSize:       DefaultDownloadPartSize,
+		Concurrency:    DefaultDownloadConcurrency,
+		BufferProvider: defaultDownloadBufferProvider(),
```

Comment:
Now that we are adding additional members that are initialized at creation of the downloader, does it make sense to create a private constructor with these default values set?

```go
func newDownloader(client s3iface.S3API,  opts ...func(*Downloader) *Downloader {...}
```

The usage functions would call the private constructor then decorate/use the downloader as needed.